components/Deposit.tsx

```diff
@@ -0,0 +1,253 @@
+import { client } from '@/components/penumbra';
+import { useQuestStore } from '@/components/store';
+import { ViewService } from '@penumbra-zone/protobuf';
+import { ValueView } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import type { CommitmentSource_Ics20Transfer } from '@penumbra-zone/protobuf/penumbra/core/component/sct/v1/sct_pb';
+import { AddressView } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import type { NotesResponse } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { AddressViewComponent } from '@penumbra-zone/ui/AddressViewComponent';
+import { ValueViewComponent } from '@penumbra-zone/ui/ValueViewComponent';
+import { useQuery } from '@tanstack/react-query';
+import { capitalize } from 'es-toolkit';
+import { ChevronRightIcon } from 'lucide-react';
+import React, { useState } from 'react';
+import {
+  useConnect,
+  useCurrentChainStatus,
+  useEphemeralAddress,
+  useNotes,
+  useSetScanSinceBlock,
+  useSwaps,
+  useWalletManifests,
+} from './hooks';
+
+const Deposit: React.FC = () => {
+  useSetScanSinceBlock();
+  const [showOld, setShowOld] = useState(false);
+
+  const { data } = useNotes();
```

Comment:
suggestion: `useNotes()` and `useSwaps()` hooks lack a fallback mechanism if the request fails. It'd be prudent to handle the error state for react query:

```
const { data, error, isLoading } = useNotes();
// display error message {error}
```

Reply:
Yes, I plan on detecting whether wallet is connected, since that's realistically the only thing besides a connection error that would cause this to fail.